src/types/relocatable.rs

```diff
@@ -349,17 +311,31 @@ impl<'a> Add<usize> for &'a Relocatable {
     }
 }
 
-/// Turns a MaybeRelocatable into a BigInt value.
-/// If the value is an Int, it will extract the BigInt value from it.
+/// Turns a MaybeRelocatable into a Felt value.
+/// If the value is an Int, it will extract the Felt value from it.
 /// If the value is Relocatable, it will return an error since it should've already been relocated.
 pub fn relocate_value(
     value: MaybeRelocatable,
     relocation_table: &Vec<usize>,
-) -> Result<BigInt, MemoryError> {
+) -> Result<Felt, MemoryError> {
     match value {
         MaybeRelocatable::Int(num) => Ok(num),
         MaybeRelocatable::RelocatableValue(relocatable) => {
-            BigInt::from_usize(relocate_address(relocatable, relocation_table)?)
+            let (segment_index, offset) = if relocatable.segment_index >= 0 {
+                (
+                    relocatable.segment_index as usize,
+                    relocatable.offset as usize,
+                )
+            } else {
+                return Err(MemoryError::TemporarySegmentInRelocation(
+                    relocatable.segment_index,
+                ));
+            };
+
+            if relocation_table.len() <= segment_index {
+                return Err(MemoryError::Relocation);
+            }
+            Felt::from_usize(relocation_table[segment_index] + offset)
```

Comment:
How and why did this grow this much?

Reply:
Resolved: https://github.com/lambdaclass/cairo-rs/pull/630/commits/a9f09a2357d79a5155251b29e93c86cf4cbcdffa